### DISCUSSION TITLE
Advice needed for using Renovate with large monorepo

### DISCUSSION THREAD

**Discussion by ngshiheng** [renovatebot/renovate#13714] [Request Help]
I am a super happy user of Renovate. However, I recently hit road block:

**Some background:**
- Golang project
- Private project on self-hosted GitLab EE
- With standard full `git clone`, the Go monorepo comes with the size of 8 GB

**Problem:**
- I ran into `Disk space error` with our GitLab runner
- I believe this is due to the huge size of monorepo that I'm working with

**Logs:**
_from my GitLab runner..._
```sh
ERROR: Disk space error - skipping (repository=ngshiheng/my-go-monorepo)
DEBUG: Unknown res (repository=ngshiheng/my-go-monorepo)
       "res": "disk-space"
DEBUG: Repository timing splits (milliseconds) (repository=ngshiheng/my-go-monorepo)
       "splits": {},
       "total": 3526038
DEBUG: http statistics (repository=ngshiheng/my-go-monorepo)
       "urls": {"https://gitlab.xxxxxxxx.net/api/v4/projects/ngshiheng%2Fgo (GET)": 1},
       "hostStats": {
         "gitlab.xxxxxxxx.net": {
           "requestCount": 1,
           "requestAvgMs": 175,
           "queueAvgMs": 0
         }
       },
       "totalRequests": 1
 INFO: Repository finished (repository=ngshiheng/my-go-monorepo)
       "durationMs": 3526038
DEBUG: Renovate exiting
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "j12usidhfaiuhs",
           "repository": "ngshiheng/my-go-monorepo",
           "msg": "Disk space error - skipping"
         }
       ]
Cleaning up file based variables
00:01
ERROR: Job failed: exit code 1
```

**Related issues that I've read:**
- https://github.com/renovatebot/renovate/issues/7840
- https://github.com/renovatebot/app-support/issues/102
- https://github.com/renovatebot/renovate/discussions/8460

After reading these related issues, I still can't seem to find the right answer to my issue.

Is there a better way around this because technically I don't need to clone all the other files that has nothing to do with my Go package manager (`go.mod`).

**What I've considered:**
Provision GitLab runners with more disk space.

**Question:**
- Is there something I am doing wrong here or is this just an infrastructure problem?
- Is there anything else that I can do/try? Workaround? 
- Is Renovate feasible for such large (in the realms of gigabytes) repo? 
- How are other people doing this?

Thanks!

**Comment by rarkins** [comment]
The problem sounds simple? Your repo size is bigger than your disk size.

You can either:
- Increase disk size (I think the cost of a few more GB would be 0.1% of the value you get from using Renovate)
- Wait for #7840 (or help move it forward)

**Reply by rarkins** [reply]
BTW #7840 is a good idea even if you can afford more than 8GB disk, because it will decrease the clone and scan times if there are directories you can completely ignore.

**Reply by ngshiheng** [reply]
Yeah that makes sense, really just wanted to confirm if I'm missing anything else. Thanks for the quick response!

One more question, do you happen to have a timeline when https://github.com/renovatebot/renovate/issues/7840 would be officially available?

**Reply by rarkins** [reply]
There is no hidden timeline, timetable or other - everything you see is in this repo. So, you are welcome to contribute towards it in any way, because it's not being worked on right now.
